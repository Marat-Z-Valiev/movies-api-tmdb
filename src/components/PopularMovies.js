import React, {useState, useEffect} from "react";
import axios from "axios";
import Results from "./Results";
import Spinner from "./Spinner";

const PopularMovies = () => {
	let [popularMovies, setPopularMovies] = useState([]);
	let [isLoading, setIsLoading] = useState(true);

	const getPopular = async () => {
		await axios
			.get(
				`https://api.themoviedb.org/3/movie/popular?api_key=${process.env.API_KEY}&language=en-US&page=1`
			)
			.then((response) => {
				setPopularMovies((popularMovies = response.data.results));
				setIsLoading((isLoading = false));
			})
			.catch((err) => console.log(`this is error ${err}`));
	};

	useEffect(() => {
		getPopular();
	}, []);

	return (
		<>
			{isLoading ? (
				<Spinner />
			) : (
				<>
					<h1 style={{textAlign: "center", color: "#ffffff"}}>
						Popular Movies
					</h1>
					<Results results={popularMovies} />
				</>
			)}
		</>
	);
};

export default PopularMovies;
