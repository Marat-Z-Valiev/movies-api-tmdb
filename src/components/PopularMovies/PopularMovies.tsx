import React, {useEffect} from "react";
import {useDispatch, useSelector} from "react-redux";
import Results from "../Results/Results";
import Spinner from "../Spinner/Spinner";
import Error from "../Error/Error";
import {
	fetchPopularMovies,
	popularMoviesSelector,
} from "../../store/slices/popularMovies";

const PopularMovies = () => {
	const dispatch = useDispatch();
	const {loading, popularMovies, hasErrors} = useSelector(
		popularMoviesSelector
	);

	useEffect(() => {
		dispatch(fetchPopularMovies());
	}, [dispatch]);

	return (
		<>
			{hasErrors ? <Error /> : ""}
			{loading ? (
				<Spinner />
			) : (
				<>
					<h1 style={{textAlign: "center", color: "#ffffff"}}>
						Popular Movies
					</h1>
					<Results results={popularMovies} />
				</>
			)}
		</>
	);
};

export default PopularMovies;
