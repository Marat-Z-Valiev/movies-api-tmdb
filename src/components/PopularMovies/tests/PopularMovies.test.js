import React from "react";
import {shallow, configure} from "enzyme";
import toJSON from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import PopularMovies from "../PopularMovies";

configure({adapter: new Adapter()});

describe("PopularMovies", () => {
	it("should render correctly", () => {
		const wrapper = shallow(<PopularMovies />);
		expect(toJSON(wrapper)).toMatchSnapshot();
	});
});
