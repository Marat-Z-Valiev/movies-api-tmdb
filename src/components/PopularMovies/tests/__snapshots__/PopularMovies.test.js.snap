// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopularMovies should render correctly 1`] = `
<Fragment>
  <Spinner />
</Fragment>
`;
